numpy/random/_bounded_integers.pyx.in

```diff
@@ -314,10 +314,15 @@ cdef object _rand_{{nptype}}(object low, object high, object size,
     high_arr = <np.ndarray>np.array(high, copy=False)
     low_ndim = np.PyArray_NDIM(low_arr)
     high_ndim = np.PyArray_NDIM(high_arr)
+    # nschloe, 2019-11-29:
+    # This conditional allows x and [x]. Why stop there? [[[[[x]]]]] also wants to be a
+    # low/high. It'd be more consistent if we just allowed int input for low, high in
+    # the first place, not array_like.
```

Comment:
I'd be inclined to just change the test here, rather than trying to allow an array

Reply:
The documentation says: `low : int or array-like` (likewise `high`). Should I change that?

Reply:
Yes

Reply:
Perhaps remove this weirdness in a separate pr?

Reply:
Ah, hang on - this is just an optimisation. There's a full broadcast path below. I'd just remove this path for all but the 0d case

Reply:
Right now, it's possible to do `randint([0, 1], [4, 7])` which gives two ints, one between 0 and 4 and another between 1 and 7. This has been introduced sometime after 1.16. I've personally never used this but I'm not sure if this is a common use case. We could allow in arrays of arbitrary shape and have the output of shape `low_shape + size` (`low_shape == high_shape` of course).

Reply:
> I'd just remove this path for all but the 0d case

Indeed.